### ISSUE TITLE
Peer dependencies for build plugins

### ISSUE THREAD

**Issue by klaussner** [meteor/meteor-feature-requests#164]
It would be helpful if build plugins could depend on npm packages from the application's `node_modules`. That would have two advantages:

* Package authors wouldn't have to update their build plugins every time a new version of an underlying npm package is released.
* Users of the build plugins would be able to use new versions as soon as they are released.

Maybe this could be implemented as an API like this:

```js
Npm.peerDepends({
  'svelte': '1.x'
});
```

`Npm.peerDepends` would be similar to `Npm.depends` but instead of causing the packages to be bundled with the build plugin, it would check if the installed versions match the given version constraints and output an error if they don't. The required npm packages would have to be installed as a `devDependency` on the application level:

```sh
meteor npm install --save-dev svelte
```

**Comment by zimme** [comment]
> The required npm packages would have to be installed as devDependency

Wouldn't it be better if it would check both `dependencies` and `devDependencies` so this could be used by other packages that isn't build plugins?

**Comment by klaussner** [comment]
Good point! Then it would be a replacement for [`checkNpmVersions`](https://github.com/tmeasday/check-npm-versions), I think.

**Comment by zimme** [comment]
I guess the question becomes, should we just be using `checkNpmVersions` instead or do we need this in Meteor?

**Comment by zimme** [comment]
I guess it would be nice to have both `peerDepends` and `devDepends` alongside `depends`.

**Comment by klaussner** [comment]
The problem is that build plugins can't access packages from `node_modules`. `checkNpmVersions` just reads the `package.json` and compares version numbers, so that doesn't help in this case.

What would `devDepends` be used for?

**Comment by zimme** [comment]
> What would devDepends be used for?

For creating a meteor package that depends on linting/project meta stuff maybe? I didn't really have a use-case I was just thinking out loud.

**Comment by mitar** [comment]
Related: https://github.com/meteor/meteor/issues/8905

**Comment by arggh** [comment]
Because of this restriction on Meteor, I had to fork @klaussner 's package `svelte:compiler` just so I can keep my app's version of `svelte` in sync with the compiler's version - and also just to keep the version up to date!

**Comment by CaptainN** [comment]
I created a [PR for meteor-svelte](https://github.com/meteor-svelte/meteor-svelte/pull/28) which uses `findUp` to grab the `package.json` file (that part was already in there!), and figure out where the `node_modules` directory is, then loads specific packages it finds in `package.json` from `node_modules` (`svelte` in this case).

The technique could be turned into some useful utility method (maybe `Npm.peerDepends`).
